src/mudlet-lua/lua/GUIUtils.lua

```diff
@@ -1156,3 +1156,158 @@ do
+
+local colours = {
+  [0] = {0,0,0}, -- black
+  [1] = {128,0,0}, -- red
+  [2] = {0,179,0}, -- green
+  [3] = {128,128,0}, -- yellow
+  [4] = {0,0,128}, --blue
+  [5] = {128,0,128}, -- magenta
+  [6] = {0,128,128}, -- cyan
+  [7] = {192,192,192}, -- white
+}
+
+local lightColours = {
+  [0] = {128,128,128}, -- black
+  [1] = {255,0,0}, -- red
+  [2] = {0,255,0}, -- green
+  [3] = {255,255,0}, -- yellow
+  [4] = {0,0,255}, --blue
+  [5] = {255,0,255}, -- magenta
+  [6] = {0,255,255}, -- cyan
+  [7] = {255,255,255}, -- white
+}
+
+-- black + 23 tone grayscale up to white
+-- The values are to be used for each of te r, g and b values
+local grayscaleComponents = {
+  [0]  =   0,
+  [1]  =  11,
+  [2]  =  22,
+  [3]  =  33,
+  [4]  =  44,
+  [5]  =  55,
+  [6]  =  67,
+  [7]  =  78,
+  [8]  =  89,
+  [9]  = 100,
+  [10] = 111,
+  [11] = 122,
+  [12] = 133,
+  [13] = 144,
+  [14] = 155,
+  [15] = 166,
+  [16] = 177,
+  [17] = 188,
+  [18] = 200,
+  [19] = 211,
+  [20] = 222,
+  [21] = 233,
+  [22] = 244,
+  [23] = 255
+}
+
+local ansiPattern = rex.new("\\e\\[([0-9;]+?)m")
+-- function for converting raw ANSI string into something decho can process
+-- bold, italics, underline not currently supported since decho doesn't support them
+function ansi2decho(text)
+  -- match each set of ansi tags, ie [0;36;40m and convert to decho equivalent.
+  -- this works since both ansi colours and echo don't need closing tags and map to each other
+  local result = rex.gsub(text, ansiPattern, function(s)
+    local output = {} -- assemble the output into this table
+
+    local t = string.split(s, ";") -- split the codes into an indexed table
+
+    -- given an xterm256 index, returns an rgb string for decho use
+    local function convertindex(tag)
+      local floor = math.floor
+      -- code from Mudlets own decoding in TBuffer::translateToPlainText
+
+      local rgb
+      if tag < 8 then
+        rgb = colours[tag]
+      elseif tag < 16 then
+        rgb = lightColours[tag-8]
+      elseif tag < 232 then
+        tag = tag - 16 -- because color 1-15 behave like normal ANSI colors
+
+        r = floor(tag / 36)
+        g = floor((tag-(r*36)) / 6)
+        b = floor((tag-(r*36))-(g*6))
+        rgb = {r*51, g*51, b*51}
+      else
+        local component = grayscaleComponents[tag - 232]
+        rgb = {component, component, component}
+      end
+
+      return rgb
+    end
+
+    -- since fg/bg can come in different order and we need them as fg:bg for decho, collect
+    -- the data first, then assemble it in the order we need at the end
+    local fg,bg
+    local i = 1
+    local floor = math.floor
+    local coloursToUse = colours
+    while i <= #t do
+      local code = t[i]
+
+      if code == '0' then -- reset attributes
+        output[#output+1] = "<r>"
+        fg,bg = nil,nil
+        coloursToUse = colours
+      elseif code == "1" then -- light or bold
+        coloursToUse = lightColours
+      elseif code == "22" then -- not light or bold
```

Comment:
Did we ever think to (or need to ) handle `\e2m;` = dim (not bold, not normal) intensity?